### PR TITLE
Jetpack Cloud: Prevent restores and downloads on non-existent and ineligible activities

### PR DESCRIPTION
**Pull request by cbauerman**
#### Changes proposed in this Pull Request

* Support new single activity endpoint
* Add Error screen when navigating to `/backups/:siteSlug/restore/:rewindId` or `/backups/:siteSlug/download/:rewindId` for an activity that is not rewindable
Screen Shot 2020-05-01 at 4 02 47 PM
* Add Error screen when navigating to `/backups/:siteSlug/restore/:rewindId` or `/backups/:siteSlug/download/:rewindId` for an activity that does not exist
Screen Shot 2020-05-01 at 4 03 21 PM

#### Testing instructions

1. For a site that has real-time backup use the activity log to navigate to an activity that pre dates backups being activated.
1. Use the actions button to both "restore" and "download"
1. Verify the page matches the "is not rewindable" screenshot above
1. Add a digit to the rewind id in the url to make an invalid rewind id
1. Verify the page matches the "activity does not exist" screenshot above

### CODE REVIEW

File: client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx [hunk 1]
```diff
@@ -191,8 +165,12 @@ const BackupDownloadFlow: FunctionComponent< Props > = ( {
 		} else if ( downloadUrl !== null ) {
 			return renderReady();
 		}
-
-		return renderError();
+		return (
+			<Error
+				siteUrl={ siteUrl }
+				errorText={ translate( 'An error occurred while creating your download.' ) }
```
Comment (a8ci18n) [inline client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx:171]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'An error occurred while creating your download' )` *ES Score: 10*

**Comment by cbauerman** [comment]
> I can see both messages, when an activity is not rewindable and when we have a bad rewindable id. When I try to download/restore a full backup is okay, but when we have an activity rewindable(upload, publish a post, ....) I got the error message, for both, the download and restore views. I tested it on a RT-Backup site.

Good catch! Turns out I did not copy the logic on the backend for making sure the correct points are marked as rewindable. Everything should be good now!

**Comment by cbauerman** [comment]
> It works well now, but I found and weird weird weird issue... only in one specific full backup (apparently), when I click on Download it goes me to the expected view but instantly jumps (without interaction) to the download process view without trigger really the process, I haven't clicked on the Download button, and neither the process started, simply I see the view:
> 
> May-12-2020 10-12-25

When you get a chance could you take a look at the body of the response of `https://public-api.wordpress.com/wpcom/v2/sites/:siteId/rewind/downloads` when that happens? It is the source of the information that determines which "phase" of the download to show, and it is possible there are edge cases I have not anticipated.

**Comment by cleacos** [comment]
image

The response is exactly the same, for any backup, any date.

The request is always without parameters:

image
It seems we are not sending any id...

Same for the `rewind` endpoint

**Comment by cbauerman** [comment]
> It seems we are not sending any id...
> 
> Same for the `rewind` endpoint

That is as expected. The endpoint is a list of recent downloads/restores for a site, and is not a specific per rewindId.

Can you put the following debug lines at around L52 in `client/landing/jetpack-cloud/sections/backups/rewind-flow/download.tsx` ?

```js
	console.log( '----------rendering BackupDownloadFlow----------' );
	console.log( 'downloadId: ', downloadId );
	console.log( 'downloadUrl: ', downloadUrl );
	console.log( 'downloadProgress: ', downloadProgress );
```

**Comment by mattgawarecki** [comment]
Testing on a new site using the original testing instructions, I was able to see the correct error messages for invalid/non-rewindable activities. I was also able to successfully create a download and do a restore.

I see there's been a lot of dialog since then, but have we discovered any other cases I should test for?

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3565912

Thank you @cbauerman for including a screenshot in the description! This is really helpful for our translators.

**Comment by cleacos** [comment]
@cbauerman It works well in production. I'm thinking that I wasn't sandboxed when I tried it the last time. (sorry)

**Pull request merged**